kernel/sys_clock.c

```diff
@@ -128,6 +128,18 @@ s64_t k_uptime_get(void)
 	return __ticks_to_ms(_tick_get());
 }
 
+#ifdef CONFIG_USERSPACE
+u32_t _handler_k_uptime_get(u32_t arg1, u32_t arg2, u32_t arg3,
+			    u32_t arg4, u32_t arg5, u32_t arg6, void *ssf)
+{
+	u64_t *ret = (u64_t *)arg1;
+
+	_SYSCALL_MEMORY(ret, sizeof(*ret), 1, ssf);
```

Comment:
This is unrelated to this patch, but I'm always going back to the definition of these syscall argument validation macros to see what's the meaning of that `0` or `1` parameter.

Can a `_SYSCALL_MEMORY_WRITE()` and `_SYSCALL_MEMORY_READ()` be defined in terms of `_SYSCALL_MEMORY()` instead?  Or at the very least, have macros defined to 0 or 1 and have those passed as arguments here.

Also, all of these macros take in `ssf` as the last argument; can't it just be omitted?  Or is there a case where the ssf pointer might be created inside the handler?

Reply:
I have another patch which I haven't sent for review yet which does exactly this, and cleans up a few other things in that header file.

Reply:
@lpereira #4255

Reply:
making the ssf parameter implicit requires that it always be named "ssf" and not something else.
i'm thinking about whether we can make the handler prototype definition a macro of some kind, which would allow that. it will be done in another PR if I find a good way to do it.

Reply:
It's not a big deal if the `ssf` parameter is explicit or implicit in those validation macros.  But if what you're working on ends up being too awkward to use, it might be better to use the proper type for it and do some type checking at least: in some macros, it's the last argument, and in some, it's the second argument.

Maybe something like this, that already takes care of the `ARG_UNUSED` stuff?

```C
#define _SYSCALL_HANDLER0(name_)  \
        u32_t _handler_ ## name_(u32_t arg1 __unused, \
                                                    u32_t arg2 __unused, \
                                                    u32_t arg3 __unused, \
			                            u32_t arg4 __unused, \
                                                    u32_t arg5 __unused, \
                                                    u32_t arg6 __unused, \
                                                    void *ssf)

_SYSCALL_HANDLER0(k_uptime_get)
{
        u64_t *ret = (u64_t *)arg1;

        _SYSCALL_MEMORY_WRITE(ret, sizeof(*ret));
        *ret = _impl_k_uptime_get();
        return 0;
}
```

Reply:
that looks really good, let me give that a try